### PR TITLE
Add tests for annotations_between_events utility

### PR DESCRIPTION
**Pull request by SchellanderF**
Added `tests/test_utils.py` to cover various scenarios for `annotations_between_events`, including:
- general functionality 
- offsets
- extensions
- edge cases

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+@pytest.fixture
+def sample_events():
+    """creates a sample events array for testing"""
+    # Format: [sample, 0, event_id]
+    return np.array([[100, 0, 1], [200, 0, 2], [500, 0, 1], [600, 0, 2]])
+
+
+def test_simple_pairing(sample_events):
+    """Tests interval annotation creation with simple start/end event pairing."""
```
Comment (cbrnr) [inline tests/test_utils.py]:
```suggestion
    """Test interval annotation creation with simple start/end event pairing."""
```

File: tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        (0.0, 0.0, 1.0, 1.0),
+        (0.1, 0.0, 1.1, 0.9),
+        (0.0, 0.2, 1.0, 1.2),
+        (-0.1, 0.1, 0.9, 1.2),
+        (0.1, -0.1, 1.1, 0.8),
+        (-0.1, -0.1, 0.9, 1.0),
+    ],
+)
+def test_offsets(sample_events, start_off, end_off, expected_onset, expected_dur):
+    """Tests interval annotation creation with start/end offsets."""
```
Comment (cbrnr) [inline tests/test_utils.py]:
```suggestion
    """Test interval annotation creation with start/end offsets."""
```

File: tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    events = sample_events[:2]
+    annots = annotations_between_events(
+        events=events,
+        sfreq=100.0,
```
Comment (cbrnr) [inline tests/test_utils.py]: I would create a global `SFREQ=100.0` and use `SFREQ` throughout the tests.

**Review by cbrnr — COMMENTED** [review]
Besides the inline comments, can you also add some tests that are expected to fail or at least have some weird parameters? For example:

```python
def test_invalid_sfreq(sample_events):
    """Test that invalid sampling frequency raises an error."""
    with pytest.raises((ValueError, TypeError)):
        annotations_between_events(
            events=sample_events,
            sfreq=0.0,  # or negative
            start_events=[1],
            end_events=[2],
            annotation="Test",
        )


def test_empty_events():
    """Test behavior with empty events array."""
    events = np.array([]).reshape(0, 3)
    annots = annotations_between_events(
        events=events,
        sfreq=SFREQ,
        start_events=[1],
        end_events=[2],
        annotation="Test",
    )
    assert len(annots) == 0
```

**Comment by SchellanderF** [comment]
> Thanks @SchellanderF! I've made some minor cosmetic changes. Before I merge, do you think we should also add tests for the new function `merge_annotations` directly? Maybe some basic tests would be good.

I agree, that’s a good idea. I will add some tests for `merge_annotations`

**Comment by cbrnr** [comment]
Thanks @SchellanderF!
